### PR TITLE
[WFLY-11293]: Add attribute to model which will indicate that backup is synchronized with live server.

### PR DESCRIPTION
**Pull request by ehsavoie**
* adding a `synchronized-with-live` attribute on `/subsystem=messaging-activemq/server=default/ha-policy=replication-slave`
* adding a `synchronized-with-backup` attribute on `/subsystem=messaging-activemq/server=default/ha-policy=replication-master`

Jira: https://issues.jboss.org/browse/WFLY-11293

### CODE REVIEW

File: messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+== Overview
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-11293[WFLY-11293]
+
+=== Related Issues
+
```
Comment (tterem) [inline messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc]: Please add https://issues.jboss.org/browse/EAP7-884 here.

File: messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
```
Comment (tterem) [inline messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc]: Set me as QE contact.

File: messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
```
Comment (tterem) [inline messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc]: You've already prepared the tests so I guess this is tested by engineering, right?

File: messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc (hunk 1 above)
Comment (ehsavoie) [reply messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc]: Hum those are community docs. i don't think those EAP7 issues should be in there

Comment (tterem) [reply messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc]: We always mentioned related EAP7 issues here. For example:
https://github.com/wildfly/wildfly-proposals/blob/master/messaging/WFLY-10326_messaging_global_resource_usage.adoc
https://github.com/wildfly/wildfly-proposals/blob/master/messaging/WFLY-11829_artemis_clusters_with_http_load_balancer.adoc
We always mentioned related EAP7 issues here.

**Comment by tterem** [comment]
Please add Overview and Test Plan sections.

File: messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+=== Affected Projects or Components
+
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+* Add a boolean `synchronized-with-backup` runtime-only attribute, with a default value set to `false`, to the https://wildscribe.github.io/WildFly/17.0/subsystem/messaging-activemq/server/ha-policy/replication-master[/subsystem=messaging-activemq//server=*/ha-policy=replication-master].
```
Comment (jmesnil) [inline messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc]: Do you mean `runtime read-only`? There should be no default value as the value will always be read from Artemis runtime

File: messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+* Add a boolean `synchronized-with-live` runtime-only attribute, with a default value set to `false`, to the https://wildscribe.github.io/WildFly/17.0/subsystem/messaging-activemq/server/ha-policy/replication-slave[/subsystem=messaging-activemq//server=*/ha-policy=replication-slave].
```
Comment (jmesnil) [inline messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc]: same comment as above

File: messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+=== Hard Requirements
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Implementation Plan
+
+Expose `org.apache.activemq.artemis.api.core.management.ActiveMQServerControl#isReplicaSync` as it represents the synchronization status of the backup on each node.
```
Comment (jmesnil) [inline messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc]: What do you mean by "expose"?

Does that mean that the new `synchronized-with-live` will read its value from `ActiveMQServerControl#isReplicaSync`or do you mean something else?

Comment (ehsavoie) [reply messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc]: yes

File: messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc (hunk 4 above)
Comment (ehsavoie) [reply messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc]: I agree

File: messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc (hunk 5 above)
Comment (ehsavoie) [reply messaging/WFLY-11293_add_attribute_to_indicate_backup_is_synchronized.adoc]: i agree

**Pull request merged**